### ISSUE TITLE
HTTP header parsing is fundamentally wrong (in parseResponseHeaders)

### ISSUE THREAD

**Issue by jbebe** [webdriverio-community/wdio-intercept-service#240]
**wdio-intercept-service**
**version 4.2.1**

This is the header parser:

```javascript
arr.forEach((header) => {
  const match = header.match(/^(.+)?:\s?(.+)$/);
  if (match) {
    headers[match[1].toLowerCase()] = match[2];
  }
});
```

This is the parsed header output for a simple server response:

```json
{
  "access-control-allow-origin: https": "//google.com/"
  "content-security-policy: default-src 'self' https": "//google.com"
  "date: wed, 23 feb 2022 14:09": "06 GMT"
  "etag: '05a73f427d81": "'0'"
  "last-modified: mon, 21 feb 2022 09:19": "00 GMT"
}
```

This lib should help you lift the burden of parsing headers: 
[parse-headers](https://www.npmjs.com/package/parse-headers)

**Comment by chmanie** [comment]
PRs are very welcome :)

Thanks for reporting, it was definitely not a good idea to attempt that from the hip there.

**Comment by jbebe** [comment]
I don't have the authority to decide if you can use one more lib. But you say it's allowed, sure, I create a PR with that npm package.

**Comment by chmanie** [comment]
It's definitely allowed. My approach was only good for some cases, I'm actually surprised that it didn't blow up much earlier

**Comment by jbebe** [comment]
> I'm actually surprised that it didn't blow up much earlier

That was my thought, too. Alright. I'll be back with a PR tomorrow.

**Comment by tehhowch** [comment]
@christian-bromann this issue can also be closed with the release of 4.3.0

**Comment by tehhowch** [comment]
Closed in #241
